postgresql/resource_postgresql_database.go

```diff
@@ -263,7 +264,18 @@ func resourcePostgreSQLDatabaseDelete(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
-	sql := fmt.Sprintf("DROP DATABASE %s", pq.QuoteIdentifier(dbName))
+	// Terminate all active connections and block new one
+	if err := terminateBConnections(c, dbName); err != nil {
```

Comment:
Both `pg_terminate_backend` and `WITH FORCE` should be an option defaulted to false. Not sure everyone want their database to be destroyed if clients are still connected on it (especially as it was not the case before).

Reply:
i mean there is lifecycle already implemented in terraform resources so if you already destroying resources we should do everything we can to remove them, if its production database or something you should ensure by lifecycle policy to don't remove it. 

If you using any kind of proxy (pgbounder, pgpool) there will be always user connected to your database so if you already executing destroying resources and there are users terraform will fail, you will need modify it (resource) and destroy it again.

Reply:
What do you mean by "ensure by lifecycle policy to don't remove it"?

Reply:
add:

  lifecycle {
    prevent_destroy = true
  }

to resource, this will ensure resource will never get deleted by terraform.

Reply:
so do you thikn @cyrilgdn  we can keep it with enforcing as terraform by default is giving us lifecycle ?